### PR TITLE
feat(): e2e testing

### LINKED ISSUE #13: e2e testing for the ocibuilder
Add and end to end testing harness to the ocibuilder to be run automatically as part of the pipeline.
Alternatively - investigate the use of Ginkgo as documented [here](https://medium.com/@william.la.martin/acceptance-testing-your-go-cli-d992926bcfe5)

### PR DESCRIPTION
**Pull request by artbegolli**
closes #13 

Includes changes from #39 . Commits **need** to be squashed.

### CODE REVIEW

File: ocictl/cmd/pull.go [hunk 1]
```diff
@@ -86,17 +86,21 @@ func (p *pullCmd) run(args []string) error {
...
 				return err
 			}
 
+			log.WithField("responses", len(res)).Debugln("received responses and running pull")
 			for idx, imageResponse := range res {
 				log.WithField("step: ", idx).Infoln("running pull step")
-				err := utils.OutputJson(imageResponse)
-				if err != nil {
+
+				if err := utils.OutputJson(imageResponse); err != nil {
```
Comment (michael-bowen-sc) [inline ocictl/cmd/pull.go]: ✅

File: pkg/apis/ocibuilder/v1alpha1/types.go [hunk 2]
```diff
@@ -375,3 +372,9 @@ type Command struct {
 	// +listType=map
 	Value []string `json:"value" protobuf:"bytes,7,opt,name=value"`
 }
+
+// Represents build image metadata
+type ImageMeta struct {
```
Comment (michael-bowen-sc) [inline pkg/apis/ocibuilder/v1alpha1/types.go]: Unless it is an industry standard acronym, we should spell out variables. 

`ImageMeta`
to
`ImageMetadata`

File: pkg/docker/docker.go [hunk 3]
```diff
@@ -53,8 +54,12 @@ func (d Docker) Build(spec v1alpha1.OCIBuilderSpec) ([]io.ReadCloser, error) {
 
 	var buildResponses []io.ReadCloser
 	for _, opt := range buildOpts {
-
 		ctx, err := common.ReadContext(opt.Context)
+		log.WithFields(logrus.Fields{
+			"localContext": opt.Context.LocalContext,
+			"gitContext": opt.Context.GitContext,
+			"s3Context": opt.Context.S3Context,
```
Comment (michael-bowen-sc) [inline pkg/docker/docker.go]: Formatting?

File: testing/e2e/resources/go-test-service/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+build:
```
Comment (michael-bowen-sc) [inline testing/e2e/resources/go-test-service/spec.yaml]: We'll need to rename this to `ocibuilder.yaml` after https://github.com/ocibuilder/ocibuilder/pull/55 is merged

Comment (artbegolli) [reply testing/e2e/resources/go-test-service/spec.yaml]: done

File: pkg/docker/docker.go (hunk 3 above)
Comment (artbegolli) [reply pkg/docker/docker.go]: done

File: pkg/apis/ocibuilder/v1alpha1/types.go (hunk 2 above)
Comment (artbegolli) [reply pkg/apis/ocibuilder/v1alpha1/types.go]: This came in from #39 - will make the change there.

**Pull request merged**